service/service_construction.go

```diff
@@ -272,23 +270,50 @@ func (s ConstructionService) ConstructionParse(
 	if !ok {
 		return nil, wrapError(errInvalidInput, fmt.Errorf("%s is not a valid address", tx.To))
 	}
+	var opMethod string
+	var currency *types.Currency
+	var value *big.Int
+	// Erc20 transfer
+	if len(tx.Data) != 0 {
+		if !mapper.EqualFoldContains(s.config.TokenWhiteList, tx.To) {
+			return nil, wrapError(errInvalidInput, "unsupported contract address associated with transaction")
+		}
+
+		toAddress, amountSent, err := parseErc20TransferData(tx.Data)
+		if err != nil {
+			return nil, wrapError(errInvalidInput, err)
+		}
+
+		contractInfo, err := s.client.ContractInfo(*toAddress, true)
```

Comment:
Parsing must be entirely offline. You should wrap the tx to sign with whatever logic is needed to parse it: https://www.rosetta-api.org/docs/construction_api_introduction.html#completely-offline